src/tests/userDetails.test.js

```diff
@@ -0,0 +1,298 @@
+import sequelize from '../database/config/db';
+import { userServices } from '../services';
+import redisClient from '../helpers/redis.js';
+import { isAuthenticated } from '../middleware/index.js';
+
+const { expect } = chai;
+chai.use(chaiHttp);
+
+const userInfo = {
+  email: 'testing@example.com',
+  password: 'Qwert@12345',
+};
+const testuser1 = {
+  id: '123',
+  displayName: 'John Doe',
+  email: 'johndoe@example.com',
+};
+describe('User Details Controller', function () {
+  let token;
+  let user;
+  before(async function () {
+    user = await userServices.getUserByEmail(userInfo.email);
+    const res = await chai.request(app).post('/users/login').send(userInfo);
+    token = res.body.token;
+  });
+
+  after(async function () {
+    await UserDetailsModel.destroy({ where: { userId: user.id } });
+    await User.destroy({ where: { id: user.id } });
+  });
+  describe('POST /users/profile/:id', function () {
+    it('should create new user details when valid data is provided', async function () {
+      const res = await chai
+        .request(app)
+        .post(`/users/profile/${user.id}`)
+        .set({ Authorization: `Bearer ${token}` })
+        .send({
+          gender: 'Male',
+          currency: 'USD',
+          lang: 'en',
+          dateOfBirth: '1990-01-01',
+          placeOfLiving: 'New York',
+          tel: '+1234567890',
+          postalCode: '0000',
+          accountNumber: '1234',
+          accountName: 'Equity',
+          country: 'Rwanda',
+          streetAdress: 'kn109',
+        });
+      expect(res).to.have.status(200);
+      expect(res.body).to.have.property('message', 'Successfully updated');
+      expect(res.body).to.be.a('object');
+    });
+    it('should update user details that previously added)', async function () {
+      const res = await chai
+        .request(app)
+        .post(`/users/profile/${user.id}`)
+        .set({ Authorization: `Bearer ${token}` })
+        .send({
+          gender: 'Male',
+          currency: 'USD',
+          lang: 'en',
+          dateOfBirth: '1990-01-01',
+          placeOfLiving: 'New York',
+          tel: '+1234567890',
+          postalCode: '44444',
+          accountNumber: '1234',
+          accountName: 'Equity test',
+          country: 'Rwanda test',
+          streetAdress: 'test',
+        });
+
+      expect(res).to.have.status(200);
+      expect(res.body).to.have.property('message', 'Successfully updated');
+      expect(res.body).to.be.a('object');
+    });
+
+    it('should update user details if some optional data are not there)', async function () {
+      const res = await chai
+        .request(app)
+        .post(`/users/profile/${user.id}`)
+        .set({ Authorization: `Bearer ${token}` })
+        .send({
+          gender: 'Male',
+          currency: 'USD',
+          lang: 'en',
+          dateOfBirth: '1990-01-01',
+          placeOfLiving: 'New York',
+          tel: '+1234567890',
```

Comment:
Improve Code readability by sending one object instead of a whole lot of key-value pairs.